analyzer/src/main/kotlin/managers/Yarn2.kt

```diff
@@ -126,21 +192,530 @@ class Yarn2(
+
+            val allProjects = parseAllPackages(iterator, definitionFile)
+            return allProjects.values.map { project ->
+                val scopeNames = listOf(DependencyType.DEPENDENCIES, DependencyType.DEV_DEPENDENCIES).map { it.type }
+                ProjectAnalyzerResult(project.copy(scopeNames = scopeNames.toSortedSet()), sortedSetOf())
+            }.toList()
+        }
+    }
+
+    /**
+     * Query the details of several package. [iterator] should come from a NDJSON file. For each package defined in this
+     * file, its header are parsed and put in [packagesHeaders].
+     * Then the packages are separated in chunks and queried with `npm file` in [workingDir]. Unfortunately, under the
+     * hood, NPM does a request per package. However, if a solution to batch these requests arise, the code is ready
+     * for it. From the response to `npm file`, package details are extracted and put in [packagesDetails].
+     */
+    private fun queryPackageDetails(iterator: MappingIterator<ObjectNode>, workingDir: File) {
+        log.info { "Parsing packages headers..." }
+
+        iterator.asSequence().forEach { json ->
+            val value = json["value"].textValue()
+            val nameMatcher = requireNotNull(EXTRACT_FROMLOCATOR_PATTERN.matchEntire(value)) {
+                "Name of package $value cannot be parsed"
+            }
+            val rawName = nameMatcher.groupValues[1]
+            val type = nameMatcher.groupValues[2] // either "npm" or "workspace"
+            val version = nameMatcher.groupValues[3]
+            packagesHeaders += value to PackageHeader(rawName, type, version)
+        }
+
+        log.info { "Fetching packages details..." }
+
+        val chunks = packagesHeaders.filterValues { it.type != "workspace" }.values.map {
+            "${it.rawName}@${cleanYarn2VersionString(it.version)}"
+        }.chunked(BULK_DETAILS_SIZE)
+
+        packagesDetails += runBlocking(Dispatchers.IO) {
+            chunks.mapIndexed { index, chunk ->
+                async {
+                    Yarn2.log.info { "Fetching packages details chunk #$index." }
+
+                    val process = run(
+                        "npm",
+                        "info",
+                        "--fields",
+                        "description,repository,dist,homepage,author,gitHead,version",
+                        "--json",
+                        *chunk.toTypedArray(),
+                        workingDir = workingDir,
+                        environment = mapOf("NODE_TLS_REJECT_UNAUTHORIZED" to "0")
+                            .takeIf { disableRegistryCertificateVerification }
+                            .orEmpty()
+                    )
+                    val parser = jsonMapper.createParser(process.stdout)
+                    parser.use {
+                        val detailsIterator =
+                            jsonMapper.readValues(parser, ObjectNode::class.java)
```

Comment:
Again inline "parser" and use "it", or use `.use { parser ->` (the style should be aligned with the other cases in this file).